### ISSUE TITLE
Sphinx-1.5.1 package contains untracked files

### ISSUE THREAD

**Issue by wallrj** [sphinx-doc/sphinx#3256]
### Problem
- I create a DEB package for Flocker which  contains Sphinx.
- I run `lintian` on the resulting package and it fails with an error.

See https://lintian.debian.org/tags/macos-ds-store-file-in-package.html

#### Procedure to reproduce the problem
```
$ git clone --depth 1 --branch 1.5.1 https://github.com/sphinx-doc/sphinx.git Sphinx-1.5.1
$ wget https://pypi.python.org/packages/b2/d5/bb4bf7fbc2e6b85d1e3832716546ecd434632d9d434a01efe87053fe5f25/Sphinx-1.5.1.tar.gz
$ tar xf Sphinx-1.5.1.tar.gz
$ cd Sphinx-1.5.1
$ git status --ignored
Not currently on any branch.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   setup.cfg

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	PKG-INFO
	"tests/roots/test-image-glob/testima\314\210ge.png"

Ignored files:
  (use "git add -f <file>..." to include in what will be committed)

	Sphinx.egg-info/
	sphinx/locale/.DS_Store
	sphinx/locale/__init__.pyc
	sphinx/pycode/Grammar-py2-sphinx1.2.pickle
	sphinx/pycode/Grammar-py2-sphinx1.3.pickle
	sphinx/pycode/Grammar-py2-sphinx1.4.pickle
	sphinx/pycode/Grammar-py2-sphinx1.5.pickle
	sphinx/pycode/Grammar-py2.pickle
	sphinx/pycode/Grammar-py3-sphinx1.4.pickle
	sphinx/pycode/Grammar-py3.pickle
	utils/regression_test.js

no changes added to commit (use "git add" and/or "git commit -a")
```

#### Error logs / results
```
...
10:45:31 {"exception": "exceptions.SystemExit", "task_level": [2], "action_type": "packaging:buildsequence:run", "reason": "1", "timestamp": 1481971389.990956, "task_uuid": "1ace8124-2a64-4d4e-b75b-a9f7a9c99e63", "action_status": "failed"}
10:45:31 Package errors (clusterhq-python-flocker):
10:45:31 W: clusterhq-python-flocker: macos-ds-store-file-in-package opt/flocker/lib/python2.7/site-packages/sphinx/locale/.DS_Store
10:45:32 {"exception": "exceptions.SystemExit", "task_level": [2], "action_type": "packaging:buildsequence:run", "reason": "1", "timestamp": 1481971391.0479, "task_uuid": "e9f4c969-1b57-4445-90db-e20ae765462a", "action_status": "failed"}
...
```
- http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/FLOC-4513-upstream-mimic/view/acceptance/job/run_acceptance_loopback_on_AWS_Ubuntu_Trusty_for_flocker.acceptance/lastFailedBuild/console

#### Expected results
- N/A

### Reproducible project / your project
- N/A

### Environment info
- OS: Ubuntu-14.04 / 16.04
- Python version: 2.7
- Sphinx version: 1.5.1
- lintian

Perhaps you could use something like https://pypi.python.org/pypi/setuptools-git to ensure that the package only contains files tracked by git.

Or add `global-exclude .DS_Store` to `MANIFEST.in` like: 
 * https://github.com/wesm/feather/blob/v0.3.1/python/MANIFEST.in

**Comment by jfbu** [comment]
I confirm the issue with `Sphinx-1.5.1.tar.gz` downloaded from https://pypi.python.org/pypi/Sphinx

```
[12:36:34] xxx:~/Downloads/Sphinx-1.5.1
Sphinx-1.5.1$ find . -name '*Store'
./sphinx/locale/.DS_Store
```

This crept in at packaging time because git repo contains no `.DS_Store`. 

```
$ git checkout 1.5.1
Note: checking out '1.5.1'.

[...]
HEAD is now at 794f6bd... Bump to 1.5.1 final

$ find . -name '*Store'

$
```

and same on master. Also `git ls-files | grep Store` finds no match.

Note: under Mac OS X, opening  a Finder window for a view of a repertory will usually create a `.DS_Store` file (I don't know the exact rules, but I hate this as it modifies repertory timestamp).

**Comment by tk0miya** [comment]
Thank you for reporting.

>	Sphinx.egg-info/

This is generated by setuptools on packaging.
@shimizukawa What is correct package?

>	sphinx/pycode/Grammar-py2.pickle
>	sphinx/pycode/Grammar-py3.pickle

They are generated on packaging. It's intended.

>	sphinx/locale/.DS_Store
>	sphinx/locale/__init__.pyc
>	sphinx/pycode/Grammar-py2-sphinx1.2.pickle
>	sphinx/pycode/Grammar-py2-sphinx1.3.pickle
>	sphinx/pycode/Grammar-py2-sphinx1.4.pickle
>	sphinx/pycode/Grammar-py2-sphinx1.5.pickle
>	sphinx/pycode/Grammar-py3-sphinx1.4.pickle
>	utils/regression_test.js

These should not be included into the package.
So we have to update the packaging script.

**Comment by shimizukawa** [comment]
> Sphinx.egg-info/

There is no specification what is a correct file layout of sdist package.
https://docs.python.org/3/library/distutils.html
https://packaging.python.org/

`Sphinx.egg-info/` is generated by setuptools when we create `sdist` package.
https://github.com/pypa/setuptools/blob/master/setuptools/command/sdist.py#L43
In fact, we don't have to contain `Sphinx.egg-info` directory to sdist package, however we can't avoid the egg-info directory even if MANIFEST.in have `prune Sphinx.egg-info` line.

**Comment by tk0miya** [comment]
Fixed at #3262.
Thank you for reporting!
